core/base-service/base-graphql.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const { parse } = require('graphql')
+const BaseJsonService = require('./base-json')
+const { InvalidResponse } = require('./errors')
+
+function defaultGraphqlErrorHandler(errors) {
+  throw new InvalidResponse({ prettyMessage: errors[0].message })
+}
+
+class BaseGraphqlService extends BaseJsonService {
+  _validateQuery(query) {
+    // Attempting to parse the query string
+    // will throw a descriptive exception if it isn't valid
+    parse(query)
+  }
+
+  async _requestGraphql({
+    schema,
+    url,
+    query,
+    variables = {},
```

Comment:
I've made `query` and `variables` top-level params here. I'm in 2 minds about whether it makes more sense to collect them into an object. Also note here I deliberately haven't implemented mutations. In general, calling a shields badge should read data from upstream service, not write data to it, so we shouldn't need mutations.